### PR TITLE
Jormungandr: refactor journeys links

### PR DESCRIPTION
**Pull request by antoine-de**
links were done in several places.

they are now done in 1 place: the scenarios

This way each scenario can use the same mechanims as the internal one to fill the journeys with the min_nb_journeys parameter

linked to http://jira.canaltp.fr/browse/NAVITIAII-2079

---

This change is [Reviewable](https://reviewable.io/reviews/canaltp/navitia/1509)

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 1]
```diff
@@ -698,3 +671,48 @@ def isochrone(self, request, instance):
...
 
         return resp
+
+    def create_next_kraken_request(self, request, responses):
+        """
+        create a new request dict to call the next (resp previous for non clockwise search) journeys in kraken
+
+        to do that we find the best journey (soonest arrival, then shortest journey) found in the responses
+        and add one second to it's departure.
+        (if anticlockwise, remove one second to the arrival of the journey with latest departure, then shortest)
```
Comment (TeXitoi) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: This comment seems deconnected from the code.

**Comment by kinnou02** [comment]
destineo is not handled!

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 2]
```diff
@@ -698,3 +671,48 @@ def isochrone(self, request, instance):
...
+        else:
+            request['datetime'] = self.previous_journey_datetime([j for r in responses for j in r.journeys])
+
+        if request['datetime'] is None:
+            return None
+
+        #TODO forbid ODTs
+        return request
+
+    def __get_best_for_criteria(self, journeys, criteria):
```
Comment (TeXitoi) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: static method?

**Comment by antoine-de** [comment]
Note: @kinnou02 

one BIG down side now is that the destineo journeys integration tests cannot run the
old default test (thus teh test class cannot inherit from `JourneyTest`) because nosetest run them as 'default' journeys, so the links checks fail (because links are different in 'destineo' and in
'default')

File: source/jormungandr/tests/routing_tests_experimental.py [hunk 3]
```diff
@@ -62,13 +63,35 @@ def teardown(self):
         from jormungandr import i_manager
         i_manager.instances['main_routing_test']._scenario = self.old_scenario
 
+    @staticmethod
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
Comment (pbougue) [inline source/jormungandr/tests/routing_tests_experimental.py]: comment is not matching code, 1mn vs 1s

File: source/jormungandr/tests/routing_tests_experimental.py [hunk 4]
```diff
@@ -62,13 +63,35 @@ def teardown(self):
...
+                                         new_default_pagination_journey_comparator(clockwise=True))
+
+        j_departure = get_valid_datetime(j_to_compare['departure_date_time'])
+        eq_(j_departure + timedelta(seconds=1), dt)
+
+    @staticmethod
+    def check_previous_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default previous behaviour is 1 mn before the best or the latest """
```
Comment (pbougue) [inline source/jormungandr/tests/routing_tests_experimental.py]: comment is not matching code, 1mn vs 1s

File: source/jormungandr/tests/routing_tests_new_default.py [hunk 5]
```diff
@@ -62,13 +62,38 @@ def teardown(self):
...
         i_manager.instances['main_routing_test']._scenario = self.old_scenario
 
+    """
+    the new default scenario override the way the prev/next link are created
+    """
+    @staticmethod
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
Comment (pbougue) [inline source/jormungandr/tests/routing_tests_new_default.py]: comment is not matching code, 1mn vs 1s

File: source/jormungandr/tests/routing_tests_new_default.py [hunk 6]
```diff
@@ -62,13 +62,38 @@ def teardown(self):
...
+                                         new_default_pagination_journey_comparator(clockwise=True))
+
+        j_departure = get_valid_datetime(j_to_compare['departure_date_time'])
+        eq_(j_departure + timedelta(seconds=1), dt)
+
+    @staticmethod
+    def check_previous_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default previous behaviour is 1 mn before the best or the latest """
```
Comment (pbougue) [inline source/jormungandr/tests/routing_tests_new_default.py]: comment is not matching code, 1mn vs 1s

File: source/jormungandr/tests/tests_mechanism.py [hunk 7]
```diff
@@ -212,6 +216,104 @@ def query_no_assert(self, url, display=False):
...
+        if query_dict.get('debug', False):
+            assert 'debug' in response
+        else:
+            assert 'debug' not in response
+
+    @staticmethod
+    def check_next_datetime_link(dt, response):
+        if not response.get('journeys'):
+            return
+        """default next behaviour is 1 mn after the best or the soonest"""
```
Comment (pbougue) [inline source/jormungandr/tests/tests_mechanism.py]: minor: 1 `min`

**Comment by pbougue** [comment]
Ok for me

**Comment by TeXitoi** [comment]
OK for me.

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**